base/preview.go

```diff
@@ -0,0 +1,47 @@
+package base
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/qri-io/qri/dsref"
+	"github.com/qri-io/qri/repo"
+)
+
+// CreatePreview generates a preview for a dataset version
+func CreatePreview(ctx context.Context, r repo.Repo, ref dsref.Ref) (ds *dataset.Dataset, err error) {
+	if ref.Path == "" {
+		return nil, fmt.Errorf("path is required")
+	}
+
+	ds, err = dsfs.LoadDataset(ctx, r.Store(), ref.Path)
+	if err != nil {
+		log.Errorf("CreatePreview loading dataset: %s", err.Error())
+		return nil, err
+	}
+
+	if err = ds.OpenBodyFile(ctx, r.Store()); err != nil {
+		log.Errorf("CreatePreview opening body file: %s", err.Error())
+		return nil, err
+	}
+
+	st := &dataset.Structure{
+		Format: "json",
+		Schema: ds.Structure.Schema,
+	}
+
+	data, err := ConvertBodyFile(ds.BodyFile(), ds.Structure, st, 100, 0, false)
```

Comment:
I would much prefer if this 100 was a well-named constant, like `defaultNumBodyRowsInPreview`, in order to call attention to it, and make it clear that it was a deliberate choice. As opposed to the couple of places in our code where we use something like 100000 to essentially mean "infinite" (see calls to `References`).